libdnf/module/ModulePackageContainer.cpp

```diff
@@ -184,6 +235,23 @@ ModulePackageContainer::ModulePackageContainer(bool allArch, std::string install
     repoImpl->libsolvRepo = repo;
     repoImpl->needs_internalizing = 1;
     pImpl->installRoot = installRoot;
+    g_autofree gchar * path = g_build_filename(pImpl->installRoot.c_str(),
+                                              "/etc/dnf/modules.d", NULL);
+    struct dirent * ent;
+    std::unique_ptr<DIR> dir(opendir(path));
+    if (dir) {
+        /* Load "*.module" files into module persistor */
+        DIR * dirPtr = dir.get();
+        while ((ent = readdir(dirPtr)) != NULL) {
+            auto filename = ent->d_name;
+            auto fileNameLen = strlen(filename);
+            if (fileNameLen < 8 || !strcmp(filename - 7, ".module")) {
```

Comment:
Access to memory before the string in `!strcmp(filename - 7, ".module")`.
Probably you wanted to write: `strcmp(filename + fileNameLen - 7, ".module")`.